app/models/manageiq/providers/openstack/cloud_manager/cloud_tenant.rb

```diff
@@ -4,4 +4,17 @@ class ManageIQ::Providers::Openstack::CloudManager::CloudTenant < ::CloudTenant
                           :join_table              => "cloud_tenants_vms",
                           :association_foreign_key => "vm_id",
                           :class_name              => "ManageIQ::Providers::Openstack::CloudManager::Template"
+
+  def self.post_refresh_ems(ems_id, _)
+    ems = ExtManagementSystem.find(ems_id)
+
+    MiqQueue.put(
```

Comment:
Although, be sure to handle race conditions that could happen if multiple refreshes complete, put multiple sync_cloud_tenants_with_tenants on the queue and then they are run concurrently.  Alternately use put_unless_exists, however, you can run into other race conditions with that approach where you could miss the sync of certain objects.  Granted another refresh will just come along, but it's not perfect.

Also, are you going to do a full sync after targeted refreshes?  (We don't have targeted refresh for OpenStack, but I'm trying to just think ahead).